app/src/components/advanceSlides.spec.js

```diff
@@ -0,0 +1,103 @@
+import { AdvanceSlides } from "./advanceSlides";
+import { store } from "../store";
+
+jest.mock("../store", () => {
+  return {
+    __esModule: true,
+    store: {
+      getState: jest.fn(() => ({
+        slides: {
+          curIndex: 0,
+        },
+      })),
+      dispatch: jest.fn(),
+      subscribe: jest.fn(),
+    },
+  };
+});
+
+describe("AdvanceSlides", () => {
+  let advanceSlides;
+  let spyButton;
+  let spyAdvanceToSlide;
+  let element;
+  const mockClickEvent = { preventDefault: jest.fn() };
+
+  beforeAll(() => {
+    setDocumentHtml(getMainHtml()); // eslint-disable-line no-undef
+    element = document.querySelector(".go-next.bottom-arrow");
+    spyButton = jest.spyOn(AdvanceSlides.prototype, "handleClick");
+    spyAdvanceToSlide = jest.spyOn(AdvanceSlides.prototype, "advanceToSlide");
+    advanceSlides = new AdvanceSlides({
+      element,
+      store,
+      buttonSelector: "h3",
+    });
+  });
+
```

Comment:
add `beforeEach` and `afterEach` here, the former should make the instance of `AdvanceSlides` and the latter should call `jest.clearAllMocks()`